### DISCUSSION TITLE
convert text to picture

### DISCUSSION THREAD

**Discussion by VladFomenko** [ImageMagick/ImageMagick#7398] [Help] [answered]
### ImageMagick version

7.1.1-33

### Operating system, version and so on

Ubuntu 22.04.4 LTS

### Description

I want to take the text and convert it into a picture, without taking into account the tags, I use pango to get emojis in the picture as well. But I have a problem with & and < if I found how to escape & and it's gsub('&', '&amp;amp;') - I write in ruby. So what to do with < I have no idea, an error always occurs. I also found -define pango:markup=false, but either it doesn't help me or I'm not using it that way. I need to simply take the text and simply translate it into a picture (but to support all fonts and emojis). Help please!!!

**Comment by VladFomenko** [comment]
```
magick -background none -fill black -font Arial -pointsize 48 "pango:<' -define pango:markup=false" combined_test.png

(process:43770): Pango-WARNING **: 13:36:00.004: pango_layout_set_markup_with_accel: Error on line 1 char 20: “-define” is not a valid name
magick: Error on line 1 char 20: “-define” is not a valid name `<' -define pango:markup=false' @ error/pango.c/ReadPANGOImage/372.
magick: negative or zero image size `<' -define pango:markup=false' @ error/image.c/SetImageExtent/2667.

```

**Reply by dlemstra** [reply, marked as answer]
The `-define pango:markup=false` should be set before `pango:`

**Comment by VladFomenko** [comment]
Thank you! It works! Now it remains to understand how to implement it in Magick::Image.read
